### PR TITLE
Added support for UnoEuro API

### PR DESCRIPTION
**Pull request by Aarup**

### CODE REVIEW

File: dnsapi/dns_unoeuro.sh [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        return 0
+      else
+        _err "Add txt record error."
+        return 1
+      fi
+    fi
+    _err "Add txt record error."
+  else
+    _info "Updating record"
+    record_id=$(echo "$response" | grep -B 1 "$_sub_domain" | _head_n -1 | _egrep_o "[0-9]{1,}")
```
Comment (Neilpang) [inline dnsapi/dns_unoeuro.sh]: Can you not use "grep -B" ?
I doubt it will be problem crossing platform.

**Comment by Aarup** [comment]
Made it using sed instead, hope that's better

File: dnsapi/dns_unoeuro.sh [hunk 2]
```diff
@@ -65,7 +65,7 @@ dns_unoeuro_add() {
     _err "Add txt record error."
   else
     _info "Updating record"
-    record_id=$(echo "$response" | grep -B 1 "$_sub_domain" | _head_n -1 | _egrep_o "[0-9]{1,}")
+    record_id=$(echo "$response" | sed -n -e "/$_sub_domain/{x;p;d;}" -e x | _head_n -1 | _egrep_o "[0-9]{1,}")
```
Comment (Neilpang) [inline dnsapi/dns_unoeuro.sh]: sorry, "sed -e" is also not good.

**Comment by Aarup** [comment]
Gave it another go, my lack of experience in shell scripting is starting to show

File: dnsapi/dns_unoeuro.sh [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+      else
+        _err "Add txt record error."
+        return 1
+      fi
+    fi
+    _err "Add txt record error."
+  else
+    _info "Updating record"
+    record_line_number=$(echo "$response" | grep -n "$_sub_domain" | cut -d : -f 1)
+    record_line_number=$((record_line_number - 1))
```
Comment (Neilpang) [inline dnsapi/dns_unoeuro.sh]: do not use "$(())",  please use "_math()" function instead.

**Comment by Neilpang** [comment]
please pull the latest code

File: dnsapi/dns_unoeuro.sh [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    return 1
+  fi
+  _debug _domain_id "$_domain_id"
+  _debug _sub_domain "$_sub_domain"
+  _debug _domain "$_domain"
+
+  _debug "Getting txt records"
+  _uno_rest GET "my/products/$h/dns/records"
+
+  if ! _contains "$response" "\"status\": 200" >/dev/null; then
```
Comment (Neilpang) [inline dnsapi/dns_unoeuro.sh]: do not need to use ">/dev/null"  for "_contains()" function.

File: dnsapi/dns_unoeuro.sh [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+Uno_Api="https://api.unoeuro.com/1"
+
+########  Public functions #####################
+
+#Usage: add  _acme-challenge.www.domain.com   "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+dns_unoeuro_add() {
+  fulldomain=$1
+  txtvalue=$2
+
+  Uno_Key="${Uno_Key:-$(_readaccountconf_mutable Uno_Key)}"
```
Comment (Neilpang) [inline dnsapi/dns_unoeuro.sh]: please rename "Uno_Key"  to "UNO_Key"

"Uno_User" to "UNO_User"

**Comment by Neilpang** [comment]
please update the name in the doc too.

**Pull request merged**